### PR TITLE
Move over to real retries method

### PR DESCRIPTION
**Pull request by nathanleclaire**
I noticed that on creates it would just burn through the retries very quickly, and also the errors weren't getting logged to `Debug` (saw it in @botchagalupe cast).  I'm still not sure why there are handshake / timeout issues so commonly after we've established a TCP dial works, but this is definitely a cleanup of this section if nothing else (includes sleeps etc. too now).

cc @ehazlett @ibuildthecloud @sthulb 
Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

**Comment by sthulb** [comment]
Looks good.

File: ssh/client.go [hunk 1]
```diff
@@ -63,23 +63,30 @@ func NewConfig(user string, auth *Auth) (*ssh.ClientConfig, error) {
 	}, nil
 }
 
+func dialSuccess(client *Client) func() bool {
+	return func() bool {
+		_, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", client.Hostname, client.Port), client.Config)
```
Comment (ehazlett) [inline ssh/client.go]: `if _, err := ...` ?

**Comment by ehazlett** [comment]
Minor nit otherwise looks good.

File: ssh/client.go (hunk 1 above)
Comment (nathanleclaire) [inline ssh/client.go]: Works for me

**Comment by nathanleclaire** [comment]
@ehazlett Updated!

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**